dl/src/test/scala/com/intel/analytics/bigdl/optim/PredictorSpec.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.optim
+
+import com.intel.analytics.bigdl.dataset.Sample
+import com.intel.analytics.bigdl.models.lenet.LeNet5
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.Engine
+import com.intel.analytics.bigdl.utils.RandomGenerator._
+import org.apache.spark.SparkContext
+import org.scalatest.{FlatSpec, Matchers}
+
+class PredictorSpec extends FlatSpec with Matchers {
+  "model.predict" should "be correct" in {
+    Engine.init(1, 1, true)
+    RNG.setSeed(100)
+    val sc = new SparkContext("local[1]", "DistriValidator")
+    val data = new Array[Sample[Float]](100)
+    var i = 0
+    val batchsize = 10
+    while (i < data.length) {
+      val input = Tensor[Float](28, 28).apply1(_ =>
+        RNG.uniform(0.130660 + i, 0.3081078).toFloat)
+      val label = Tensor[Float](1).fill(1.0f)
+      data(i) = Sample(input, label)
+      i += 1
+    }
+    val model = LeNet5(classNum = 10)
+    val dataSet = sc.parallelize(data, 2)
+    val result = model.predict(dataSet)
+
+    val prob = result.map(_.toTensor[Float].clone()).collect()
+    prob(0) should be (model.forward(data(0).feature()))
+    prob(11) should be (model.forward(data(11).feature()))
+    prob(31) should be (model.forward(data(31).feature()))
+    prob(51) should be (model.forward(data(51).feature()))
+    prob(71) should be (model.forward(data(71).feature()))
+    prob(91) should be (model.forward(data(91).feature()))
+
+    sc.stop()
+  }
+
+  "model.predictClass" should "be correct" in {
+    Engine.init(1, 1, true)
+    RNG.setSeed(100)
+    val sc = new SparkContext("local[1]", "DistriValidator")
+    val data = new Array[Sample[Float]](100)
```

Comment:
May be we can revise the unitest which the length of data is not divisible by node*core*4 to see if the result is correct or not.